workflow/controller/controller.go

```diff
@@ -386,13 +459,7 @@ func (wfc *WorkflowController) processNextPodItem() bool {
 
 func (wfc *WorkflowController) tweakWorkflowlist(options *metav1.ListOptions) {
 	options.FieldSelector = fields.Everything().String()
-	// completed notin (true)
-	incompleteReq, err := labels.NewRequirement(common.LabelKeyCompleted, selection.NotIn, []string{"true"})
```

Comment:
significant change

Reply:
As I understand, you removed this because you need to look at completed workflows. However we have a TTL controller which is specifically filtered against completed workflows.

The controller/logic for dealing with completed workflows, should be separate from the controller dealing with running workflows. I think the all GC logic should be moved to the TTL controller, especially because some synchronization might need to happen between them.

Reply:
As discussed - removing.